bluesky/run_engine.py

```diff
@@ -1353,7 +1353,18 @@ def _open_run(self, msg):
         self._run_start_uids.append(self._run_start_uid)
 
         # Run scan_id calculation method
-        self.md['scan_id'] = self.scan_id_source(self.md)
+        stateful_md = self.md.copy()
```

Comment:
It is a bit pedantic, but I find this variable name confusing, we are using it to specifically make sure that our changes are _not_ stateful!

To that end `local_md = dict(self.md)` may be safer (incase the `copy` method of the stateful thing returns a copy if it's self in a way that is still tied to what ever we are trying to seperate it from.  By using `dict()` we are sure we have a plain python dict).

Reply:
The "stateful" name comes from this comment: https://github.com/NSLS-II/bluesky/blob/ebc912a9962e7796d3d1b03f2d5da08804d345fd/bluesky/run_engine.py#L1367

I'm fine with making it `local_md` as suggested. +1 for the `dict()` approach.

Reply:
Implemented via https://github.com/NSLS-II/bluesky/pull/1175/commits/d4b0aa4ab8b3cf02c7223024e3e31d803f8828ef.